rllib/agents/dqn/apex.py

```diff
@@ -455,19 +472,12 @@ def remote_worker_sample_and_store(
 
         # Sample and Store in the Replay Actors on the sampling workers.
         with self._timers[SAMPLE_TIMER]:
-            # Results are a mapping from ActorHandle (RolloutWorker) to their
-            # returned gradient calculation results.
-            num_samples_ready_dict: Dict[
-                ActorHandle, T
-            ] = asynchronous_parallel_requests(
-                remote_requests_in_flight=self.remote_sampling_requests_in_flight,
-                actors=self.workers.remote_workers(),
-                ray_wait_timeout_s=0.1,
-                max_remote_requests_in_flight_per_actor=4,
-                remote_fn=remote_worker_sample_and_store,
-                remote_kwargs=[{"replay_actors": self.replay_actors}]
-                * len(self.workers.remote_workers()),
+            self._sampling_actor_manager.submit(
+                remote_worker_sample_and_store,
+                fn_kwargs={"replay_actors": self._replay_actors},
+                for_all_workers=True,
             )
+            num_samples_ready_dict = self._sampling_actor_manager.get_ready_results()
```

Comment:
1 thing to double check. if num_samples_ready_dict does not contain enough samples for the next iteration, what will happen? we will come back and fetch another batch, right?
also, probably should name this ready_samples_dict